File: DN DSA THA/Sandhya tha/Binary Search.cpp
int binaryS(vector<int> &nums, int l , int h, int target){
    int mid = l+(h-l)/2;
    if(l <= h){
       if(nums[mid]==target)
         return mid;
        else if(nums[mid]>target)
         return binaryS(nums, l, mid-1, target);
        else
          return binaryS(nums, mid+1, h, target); 
     }
    return -1;   
}

File: java/sky/LeetCode/08-08-2022/binary-search.java
//link: https://leetcode.com/problems/binary-search/

//using recursion

class Solution {
    private int bs(int[] nums, int target, int s, int e){
        if(s > e)   return -1;
        
        int m = e + (s - e)/2;
        int elm = nums[m];
        if(target == elm)
            return m;
        if(target < elm)
            return bs(nums, target, s, m-1);
        if(target > elm)
            return bs(nums, target, m+1, e);
        
        return -1;
    }
    public int search(int[] nums, int target) {
        return bs(nums, target, 0, nums.length-1);
    }
}